util/TorqueLog.java

```diff
@@ -0,0 +1,153 @@
+     */
+    public TorqueLog(final String title) {
+        this.title = title;
+        tab = Shuffleboard.getTab(title);
+        keys = new HashMap<String, NetworkTableEntry>();
+    }
+
+    /**
+     * Log a value to the dashboard.
+     * 
+     * @param key The title of the widget.
+     * @param value The value to log in the widget.
+     * @param width The width of the widget in cells.
+     * @param height The height of the widget in cells.
+     * @param type The type of the cell.
+     */
+    public void log(final String key, final Object value, final int width, final int height, final WidgetType type) {
+        if (!keys.containsKey(key)) {
+            final Position pos = calculatePosition(width, height);
+            if (pos == Position.INVALID) {
+                System.out.println("Could not fit element " + key + " in log " + title);
+                return;
+            }
+            final NetworkTableEntry entry = tab.add(key, value)
+                    .withWidget(type)
+                    .withSize(width, height)
+                    .withPosition(pos.x, pos.y)
+                    .getEntry();
+            keys.put(key, entry);
+        }
+        keys.get(key).setValue(value);
+    }
+    
+    /**
+     * Find the next coordinate location that a widget of size (width, height) can fit onto the dashboard.
+     * 
+     * This algorithm is definitely not the most efficient because it kind of checks every position
+     * on the board. However, it is simple and works well enough for our purposes. Since it only runs once
+     * on the constructor with a very small size it doesn't matter at all.
+     * 
+     * @param width The width of the widget in cells.
+     * @param height The height of the widget in cells.
+     * 
+     * @return The next valid position or INVALID if there isn't a valid position.
+     */
+    private Position calculatePosition(final int width, final int height) {
+        for (int i = 0; i < board.length; i++)
+            for (int j = 0; j < board[i].length; j++) {
+                if (board[i][j]) continue;
+                if (!widgetWillFit(i, j, width, height)) continue;
+                markWidgetTaken(i, j, width, height);
+                return new Position(i, j);
+            }
+        return Position.INVALID;
+    }
+
+    /**
+     * Check if an widget of size (width, height) will fit at the location (x, y).
+     * 
+     * @param x The x location of the widget.
+     * @param y The y location of the widget.
+     * @param width The width of the widget in cells.
+     * @param height The height of the widget in cells.
+     * 
+     * @return If the widget will fit or not.
+     */
+    private boolean widgetWillFit(final int x, final int y, final int width, final int height) {
+        for (int i = 0; i < height; i++)
+            for (int j = 0; j < width; j++)
+                if (i + x >= board.length || j + y >= board[i].length || board[i + x][j + y])
```

Comment:
If you do the comment I said above, you can reduce this if statement by not having to check if they are out of bounds